Makefile

```diff
@@ -7,3 +7,76 @@ $(TARGETS): $(SUBDIRS)
 
 $(SUBDIRS):
 	@$(MAKE) -C $@ $(MAKECMDGOALS)
+
+
```

Comment:
How do you feel about placing this codegen part in a Makefile located at `scripts/Makefile.codegen`? We can either import it or just ask users to specify the path (`make -f scripts/Makefile.codegen`) when they want to use these targets.

Reply:
As long as the developer doesn't have to remember a complex chain of commands I'm good with this.

Reply:
let create the `scripts/Makefile.codegen` and include it in `Makefile`.

Reply:
Done in 75e316b